install.py

```diff
@@ -126,7 +126,7 @@ def removeFile(f):
     try:
         os.unlink(f)
     except OSError:
-        logError('Error removing "%s".' % (f, ))
+        logError(f'Error removing "{f}".')
```

Comment:
does the syntax ` % (f, )` have a special meaning compared to ` % (f)`?

Reply:
In py2, string formatting was done by the % operator:
>>> x = 'xxx'
>>> y = 'yyy'
>>> '... %s ... %s ...' % (x,y)
'... xxx ... yyy ...'
Here (x,y) is a tuple with 2 elements. If we need only 1 element, we can use a 1-element tuple (x,)
>>> '... %s ...' % (x,)
'... xxx ...'
However, a tuple is not really necessary here, and we could simply use
>>> '... %s ...' % x
'... xxx ...'
In py3 the % operator still works, but it is kind of deprecated. Not technically deprecated, but the preferred method is to use f-strings:
>>> f'...{x}...'
'...xxx...'
It improves readability, especially in long multiline strings like
"""Usage: %s [OPTION...]
Install or remove Diffuse.
Options:
  --help
     print this help text and quit
  --remove
     remove the program
  --destdir=PATH
     path to the installation's root directory
     default: %s
  --prefix=PATH
     common installation prefix for files
     default: %s
  --sysconfdir=PATH
     directory for installing read-only single-machine data
     default: %s
  --examplesdir=PATH
     directory for example configuration files
     default: %s
  --mandir=PATH
     directory for man pages
     default: %s
  --pythonbin=PATH
     command for python interpreter
     default: %s
  --files-only
     only install/remove files; skip the post install/removal tasks""" % (app_path, options['destdir'], options['prefix'], options['sysconfdir'], options['examplesdir'], options['mandir'], options['pythonbin'])
In order to find what a specific %s will insert, one has to go to the bottom line and to count the n-th element of the tuple there. The f-string version is more readable:
f"""Usage: {app_path} [OPTION...]
Install or remove Diffuse.
Options:
  --help
     print this help text and quit
  --remove
     remove the program
  --destdir=PATH
     path to the installation's root directory
     default: {options['destdir']}
  --prefix=PATH
     common installation prefix for files
     default: {options['prefix']}
  --sysconfdir=PATH
     directory for installing read-only single-machine data
     default: {options['sysconfdir']}
  --examplesdir=PATH
     directory for example configuration files
     default: {options['examplesdir']}
  --mandir=PATH
     directory for man pages
     default: {options['mandir']}
  --pythonbin=PATH
     command for python interpreter
     default: {options['pythonbin']}
  --files-only
     only install/remove files; skip the post install/removal tasks"""
For consistency, I've replaced all % operators by the modern f-string syntax. Though it is not strictly necessary.